frontend/lib/src/components/transaction_component/transaction_component.dart

```diff
@@ -0,0 +1,178 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_form_builder/flutter_form_builder.dart';
+import 'package:form_builder_validators/form_builder_validators.dart';
+import 'package:frontend/src/components/input/input_field.dart';
+import 'package:get/get.dart';
+
+/*
+  Reusable transaction component for deposit, withdraw and transfer cash
+
+  @param label, String value to display what kind of transaction it is, 
+    basicaly something like a title.
+
+  @param transfer, bool value to tell if the transaction is transfering money.
+     to activate the email feild.
+*/
+
+class TransactionComponent extends StatefulWidget {
+  const TransactionComponent({
+    super.key,
+    required this.label,
+    required this.transfer,
+  });
+
+  final String label;
+  final bool transfer;
+
+  @override
+  State<TransactionComponent> createState() {
+    return _TransactionComponentState();
+  }
+}
+
+class _TransactionComponentState extends State<TransactionComponent> {
+  final TextEditingController amountController = TextEditingController();
+  final TextEditingController emailController = TextEditingController();
+
+  void submitAmountData() {
+    final enteredAmount = double.tryParse(amountController.text);
+    final amountIsInvalid = enteredAmount == null || enteredAmount <= 0;
```

Comment:
for amount validation, we can use `'package:flutter/services.dart'` then use inputFormatters in **FormBuilderTextField()**.

ex.
```
child: FormBuilderTextField(
              name: name,
              validator: validator,
              controller: controller,
              keyboardType: inputType,
              decoration: const InputDecoration(
                border: OutlineInputBorder(
                  borderRadius: BorderRadius.all(Radius.circular(8)),
                ),
              ),
              obscureText: obscureText ?? false,
              inputFormatters: <TextInputFormatter>[
                inputType == TextInputType.number
                    ? FilteringTextInputFormatter.digitsOnly
                    : FilteringTextInputFormatter.deny(''),
              ],
            )
```

Reply:
since there is already keyboardType that display only digit if the type is number.  and also if the user type alpha character the validation will fail or the error will diplay under the field and the button will not function.